Lib9c/Action/RapidCombination.cs

```diff
@@ -99,14 +100,51 @@ public override IAccountStateDelta Execute(IActionContext context)
                     $"context block index: {context.BlockIndex}, actionable block index : {actionableBlockIndex}");
             }
 
-            var count = RapidCombination0.CalculateHourglassCount(gameConfigState, diff);
+            int costHourglassCount = 0;
+
+            
+            if (slotState.PetId > 0)
+            {
+                var petStateAddress = PetState.DeriveAddress(avatarAddress, slotState.PetId);
+                if (!states.TryGetState(petStateAddress, out List rawState))
+                {
+                    throw new FailedLoadStateException($"{addressesHex}Aborted as the {nameof(PetState)} was failed to load.");
+                }
+
+                var petState = new PetState(rawState);
```

Comment:
Quick question: Can we reuse one pet for multiple combination slots?

We don't have any lock or validation logic whether this pet is used or not. If one pet must be used in one slot for the same time, we need to lock a pet for a while.

Reply:
No we can't. I am planning to implement it on craft actions like `CombinationEquipment`.

Reply:
Temporarily locking pet is implemented.
You can take a look at the diffs below :
[Pet is locked with blockIndex.](https://github.com/planetarium/lib9c/pull/1711/files#diff-de154bcb1f6ef8fff785c3d3b534ab9df93a32d178b73ba8bbe6cf56d296004fR48-R56)
[How to lock a pet with blockIndex](
https://github.com/planetarium/lib9c/pull/1711/files#diff-8b572fd20d45169bf372e3da483e4bdae131398a0c7fc28e62b1d82f1c648246R466-R473)
[How to check if a pet is free now](https://github.com/planetarium/lib9c/pull/1711/files#diff-8b572fd20d45169bf372e3da483e4bdae131398a0c7fc28e62b1d82f1c648246R155-R171)